openwisp_users/admin.py

```diff
@@ -139,10 +179,32 @@ def has_change_permission(self, request, obj=None):
         return super(UserAdmin, self).has_change_permission(request, obj)
 
     def get_queryset(self, request):
-        qs = super(UserAdmin, self).get_queryset(request)
-        # hide superusers from operators (they can't edit their details)
+        """
+        if operator is logged in - show only users
+        from same organization and hide superusers
+        if superuser is logged in - show all users
+        """
         if not request.user.is_superuser:
-            qs = qs.filter(is_superuser=False)
+            user = request.user
+            org_users = OrganizationUser.objects.filter(user=user, is_admin=True) \
```

Comment:
if an operator `is_admin` in org1 and **not** `is_admin` in org2
then he should be able to see users of org1 only, isn't it?

The `get_queryset()` in previous commit only checks if `is_superuser`, then takes all his organisations and shows records of all the users in all the organisations.
I have added a check for `is_admin` as well. Is my understanding of `is_admin` correct?

Reply:
@atb00ker yes although this would fix #34 as well, which I think it's ok, we can do this

Reply:
@nemesisdesign yes, thank you, #34 clears it.
I'll fix the behaviour in case:
if an operator `is_admin` in org1 and **not** `is_admin` in org2 and a member of default, org1 and org2 exist. Since, this member's record is visible to operator. The dropdown box doesn't contain the name of other organisations it looks like
![this](https://user-images.githubusercontent.com/24985760/47264510-35102180-d536-11e8-8ac4-4679711fad12.png).

According to #34, i'll simply hide all other organisation's record and only org1 will be visible! :smile:

Reply:
right.. this one we didn't consider.
This case right now shouldn't happen happen often, actually there's very little chance it will happen to any user right now, so I think we can create a new issue just for this case and go ahead, we'll live with the known bug for the moment while we think about a solution.

Reply:
right.. this one we didn't consider.
This case right now shouldn't happen happen often, actually there's very little chance it will happen to any user right now, so I think we can create a new issue just for this case and go ahead, we'll live with the known bug for the moment while we think about a solution.

Reply:
right.. this one we didn't consider.
This case right now shouldn't happen happen often, actually there's very little chance it will happen to any user right now, so I think we can create a new issue just for this case and go ahead, we'll live with the known bug for the moment while we think about a solution.

Reply:
right.. this one we didn't consider.
This case right now shouldn't happen happen often, actually there's very little chance it will happen to any user right now, so I think we can create a new issue just for this case and go ahead, we'll live with the known bug for the moment while we think about a solution.

Reply:
right.. this one we didn't consider.
This case right now shouldn't happen happen often, actually there's very little chance it will happen to any user right now, so I think we can create a new issue just for this case and go ahead, we'll live with the known bug for the moment while we think about a solution.

Reply:
right.. this one we didn't consider.
This case right now shouldn't happen happen often, actually there's very little chance it will happen to any user right now, so I think we can create a new issue just for this case and go ahead, we'll live with the known bug for the moment while we think about a solution.

Reply:
right.. this one we didn't consider.
This case right now shouldn't happen happen often, actually there's very little chance it will happen to any user right now, so I think we can create a new issue just for this case and go ahead, we'll live with the known bug for the moment while we think about a solution.

Reply:
right.. this one we didn't consider.
This case right now shouldn't happen happen often, actually there's very little chance it will happen to any user right now, so I think we can create a new issue just for this case and go ahead, we'll live with the known bug for the moment while we think about a solution.

Reply:
right.. this one we didn't consider.
This case right now shouldn't happen happen often, actually there's very little chance it will happen to any user right now, so I think we can create a new issue just for this case and go ahead, we'll live with the known bug for the moment while we think about a solution.

Reply:
This bug will always be a problem for the case of #34 (because the operator himself would see similar results for his own record). Hence, i think #34 is more relevant for this discussion.
I have not done any action relating to solving #34 here, i think it'll need more changes (and this discussion), doing them here feels like mixing things. :smile: